test/e2e/testnet/testnet.go

```diff
@@ -30,42 +30,25 @@ type Testnet struct {
 	knuu      *knuu.Knuu
 }
 
-func New(ctx context.Context, name string, seed int64, grafana *GrafanaInfo, chainID string,
-	genesisModifiers ...genesis.Modifier) (
-	*Testnet, error,
-) {
-	identifier := fmt.Sprintf("%s_%s", name, time.Now().Format("20060102_150405"))
-	kn, err := knuu.New(ctx, knuu.Options{
-		Scope:        identifier,
-		ProxyEnabled: true,
-		// if the tests timeout, pass the timeout option
-		// Timeout: 120 * time.Minute,
-	})
-	if err != nil {
-		return nil, err
-	}
-
-	log.Info().
-		Str("scope", kn.Scope).
-		Str("TestName", name).
-		Msg("Knuu initialized")
-
-	kn.HandleStopSignal(ctx)
+type Options struct {
+	Seed             int64
+	Grafana          *GrafanaInfo
+	ChainID          string
+	GenesisModifiers []genesis.Modifier
+	Knuu             *knuu.Knuu
+}
 
+func New(opts Options) (*Testnet, error) {
 	return &Testnet{
-		seed:    seed,
+		seed:    opts.Seed,
 		nodes:   make([]*Node, 0),
-		genesis: genesis.NewDefaultGenesis().WithChainID(chainID).WithModifiers(genesisModifiers...),
-		keygen:  newKeyGenerator(seed),
-		grafana: grafana,
-		knuu:    kn,
+		genesis: genesis.NewDefaultGenesis().WithChainID(opts.ChainID).WithModifiers(opts.GenesisModifiers...),
+		keygen:  newKeyGenerator(opts.Seed),
+		grafana: opts.Grafana,
+		knuu:    opts.Knuu,
 	}, nil
 }
 
-func (t *Testnet) Knuu() *knuu.Knuu {
-	return t.knuu
-}
-
```

Comment:
[question] is this getter no longer helpful?

Reply:
Not really as the user passes the knuu object, so they can use it on their end the way they need.